README.md

```diff
@@ -23,6 +23,34 @@ var log = Log.ForContext<HomeController>();
 log.Information("Info Log that also contains HttpContext request info and default tags");
 ```
 
+SetUserIdentity
+```csharp
+ public class LogUserNameMiddleware
+    {
+        private readonly RequestDelegate next;
+
+        public LogUserNameMiddleware(RequestDelegate next)
+        {
+            this.next = next;
+        }
+
+        public Task Invoke(HttpContext context)
+        {
+            if (context.User.Identity.IsAuthenticated)
```

Comment:
I think this would read much much better if we reworked this to:

```cs
using Exceptionless.Models;
using Exceptionless.Models.Data;
..,

var identity = context.User.Identity;
if (identity.IsAuthenticated)
            {
                LogContext.PushProperty(Event.KnownDataKeys.UserInfo, new UserInfo(identity.GetClaimValue("email"), identity.GetClaimValue("username")), true);
```

But it might be better if we just added an extension method for this to `LogContext` So you wouldn't even have to import the models or do any of the identity stuff. The extension method could just take the identity and check to see if the email and username is set and then set the user info. This would also greatly simplify the checks in the sink because it would only be populated if it was set when we added it.